stdlib/builtin/symbol.rbs

```diff
@@ -36,99 +34,202 @@ class Symbol < Object
-  # :fred.inspect   #=> ":fred"
-  # ```
+  #     :fred.inspect   #=> ":fred"
+  # 
   def inspect: () -> String
 
-  # In general, `to_sym` returns the `Symbol` corresponding to an object. As
-  # *sym* is already a symbol, `self` is returned in this case.
+  # In general, `to_sym` returns the Symbol corresponding to an object. As *sym*
+  # is already a symbol, `self` is returned in this case.
+  # 
   def intern: () -> self
 
-  # Same as `sym.to_s.length` .
+  # Same as `sym.to_s.length`.
+  # 
   def length: () -> Integer
 
-  def match: (untyped obj) -> Integer?
+  # Returns `sym.to_s.match`.
+  # 
+  def match: (Regexp | String pattern, ?Integer pos) ?{ (MatchData) -> void } -> MatchData?
 
-  def match?: (*untyped args) -> untyped
+  # Returns `sym.to_s.match?`.
+  # 
+  def match?: (Regexp | String pattern, ?Integer pos) -> bool
 
-  # Same as `sym.to_s.succ.intern` .
+  # Same as `sym.to_s.succ.intern`.
+  # 
   def next: () -> Symbol
 
-  # Same as `sym.to_s.succ.intern` .
-  def succ: () -> Symbol
+  # Same as `sym.to_s.length`.
+  # 
+  alias size length
 
-  # Same as `sym.to_s.swapcase.intern` .
-  def swapcase: () -> Symbol
+  # Returns `sym.to_s[]`.
+  # 
+  alias slice `[]`
 
-  # Returns a *Proc* object which responds to the given method by *sym* .
+  # Returns true if `sym` starts with one of the `prefixes` given. Each of the
+  # `prefixes` should be a String or a Regexp.
   # 
-  # ```ruby
-  # (1..3).collect(&:to_s)  #=> ["1", "2", "3"]
-  # ```
-  def to_proc: () -> Proc
+  #     :hello.start_with?("hell")               #=> true
+  #     :hello.start_with?(/H/i)                 #=> true
+  # 
+  #     # returns true if one of the prefixes matches.
+  #     :hello.start_with?("heaven", "hell")     #=> true
+  #     :hello.start_with?("heaven", "paradise") #=> false
+  # 
+  # variadic type parameter is not supported yet
+  # https://github.com/ruby/ruby-signature/issues/21
+  def start_with?: (String str, *String str) -> bool
 
-  # Same as `sym.to_s.upcase.intern` .
-  def upcase: () -> Symbol
+  # Same as `sym.to_s.succ.intern`.
+  # 
+  alias succ next
 
-  # Same as `sym.to_s.length` .
-  def size: () -> Integer
+  # Same as `sym.to_s.swapcase.intern`.
+  # 
+  def swapcase: () -> Symbol
+              | (:ascii | :lithuanian | :turkic) -> Symbol
+              | (:lithuanian, :turkic) -> Symbol
+              | (:turkic, :lithuanian) -> Symbol
 
-  def slice: (Integer idx_or_range) -> String
-           | (Integer idx_or_range, ?Integer n) -> String
-           | (::Range[Integer] idx_or_range) -> String
+  # Returns a *Proc* object which responds to the given method by *sym*.
+  # 
+  #     (1..3).collect(&:to_s)  #=> ["1", "2", "3"]
+  # 
+  def to_proc: () -> Proc
 
-  # Returns the name or string corresponding to *sym* .
+  # Returns the name or string corresponding to *sym*.
+  # 
+  #     :fred.id2name   #=> "fred"
+  #     :ginger.to_s    #=> "ginger"
+  # 
+  # # arglists 💪👽🚨 << Delete this section
```

Comment:
💪 👽 🚨

Reply:
:see_no_evil: 

 I added a test and update the signature and remove the comment.